server/app/services/applicant/ApplicantService.java

```diff
@@ -558,18 +558,23 @@ private ApplicationPrograms relevantProgramsForApplicant(
                     .build());
             programNamesWithApplications.add(programName);
           } else if (maybeSubmittedApp.isPresent() && activeProgramNames.containsKey(programName)) {
-            ProgramDefinition programDef = activeProgramNames.get(programName);
+            // When extracting the application status, the definitions associated with the program
+            // version at the time of submission are used. However, when clicking "reapply", we use
+            // the latest program version below.
+            ProgramDefinition applicationProgramVersion =
+                maybeSubmittedApp.get().getProgram().getProgramDefinition();
+            Optional<String> maybeLatestStatus = maybeSubmittedApp.get().getLatestStatus();
             Optional<StatusDefinitions.Status> maybeCurrentStatus =
-                programDef.statusDefinitions().getStatuses().stream()
-                    .filter(
-                        programStatus ->
-                            programStatus
-                                .statusText()
-                                .equals(maybeSubmittedApp.get().getLatestStatus().orElse("")))
-                    .findFirst();
+                maybeLatestStatus.isPresent()
```

Comment:
`ifPresent` requires a function that returns `void`. That means we need to assign the variable in the supplier, which led to an issue with it needing to be marked `final`. I never got it to a point where the compiler was happy. I'm not sure it would end up being more readable in the long run. Happy to switch to something else in a follow-on PR if you'd like.